### PR TITLE
Restructure JSON serialization code

### PR DESCRIPTION
**Pull request by rbergen**
This:

- Redistributes the JSON (de)serialization code to be close to the types it (de)serializes.
- As such, gets rid of serialization.h altogether.
- Changes the initialization of LEDFeature: the _canvas member is not set at construction, but when the LEDFeature is added to the canvas. For one, this ensures the LEDFeature is associated with the canvas it is part of, and it allows for straightforward deserialization from JSON. The code does ensure that the canvas for a LEDFeature is set only once (i.e., it can't be associated to two canvases).

As a wise man once said: it works in my office! :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,8 +29,8 @@ endef
 # Detect platform
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S), Darwin)
-    CFLAGS += -I$(shell brew --prefix)/include/ -I$(shell brew --prefix spdlog)/include
-    LDFLAGS += -L$(shell brew --prefix)/lib/ -L$(shell brew --prefix fmt)/lib
+    CFLAGS += -I$(shell brew --prefix)/include/
+    LDFLAGS += -L$(shell brew --prefix)/lib/
```
Comment (rbergen) [inline Makefile:33]: The CI build on Apple still works without the explicit include and lib references (i.e., spdlog and fmt do not seem to be keg-only). Please confirm this is also true on your Apple hardware.

File: monitor/content.cpp [hunk 2]
```diff
@@ -138,7 +134,7 @@ void Monitor::drawContent()
                             int queueColor = queueDepth < 100 ? 1 : 
                                            queueDepth < 250 ? 6 : 2;
 
-                            string bar = buildProgressBar(queueDepth, kFatQueue, 6);
+                            std::string bar = buildProgressBar(queueDepth, kFatQueue, 6);
```
Comment (rbergen) [inline monitor/content.cpp:137]: Explicit std:: namespace reference is necessary because we dropped the includes of "main" NDSCPP headers.

**Comment by davepl** [comment]
OK, somewhat backwards, I approved the #16 PR.  Now do you approve the #15 PR or how do we get all this code into main?

Thanks!
Dave

**Comment by rbergen** [comment]
I'll give the now-updated #15 one more look, and merge it if I think it (still) looks good.

**Pull request merged**